Tutorials/Level3Tutorial.md

```diff
@@ -0,0 +1,2583 @@
+* Implements a simple `clock` handler that is called from the SST Core
+* Prints console messages at a standard candence.
+
+#### Class Definition
+
+The first step in defining an external component is to create 
+it's constituent header file.  There are a number of key items that 
+must be defined in this header.  First, we must include the required 
+SST Core header files.  These basic headers provide the base SST classes, 
+template methods and ELI macros utilized to define an external component.  
+This is the minimum set of headers required to construct an external component.
+
+Next, we define the component's C++ namespaces and main class definition.
+The outer namespace is the `SST` namespace to include all the required 
+classes and methods provided by the SST Core infrastructure.  The inner 
+namespace is our external component, `basicComponent`.  Within the 
+`basicComponent` namespace we can define our initial component class.  
+The `basicClock` class inherits the `SST::Component` base class.  This base 
+class provides all the necessary functionality sufficient to intiaalize the 
+component, read component parameters, register the clock handlers and 
+interact with the core simulation methods.  The base class requires 
+that we define a constructor method that receives the component ID as 
+defined by the SST Core and an `SST::Params` object that contains 
+all the parameters defined for the target component object in the 
+SST simulation configuration.  We also define a basic destructor method.
+
+For the private data, the implementation is only required to define 
+a single private method as the clock handler.  This function, defined 
+here as `clock`, requires a single argument that is the current cycle 
+as defined by the SST Core simulation.  The `clock` method returns *true* 
+if the component is ready to signal completion to the SST Core, *false* otherwise.  
+
+In our implementation, we also define several optional parameters.  The first
+parameter, `out`, is an `SST::Output` object utilized to coalesce messages 
+through the SST Core to console.  This can include standard messages, warnings, 
+errors and fatal messages the trigger the simulation to stop.  The second 
+parameter, `clockFreq` is a string variable that will store the clock 
+frequency of our component as configured by the SST simulation script.  Finally, 
+we define two variables, `cycleCount` and `printInterval` as `Cycle_t` variables.  
+Cycle_t is SST's data container to hold clock cycles.  The `cycleCount` 
+variable is a counter that will be decremented every cycle until it reaches 
+zero.  In this manner, the component will execute for `cycleCount` cycles.  The 
+`printInterval` counter defines the interval at which we will print a message 
+to the console during the component simulation.
+
+```
+//
+// _basicComponent_h_
+//
+
+#ifndef _BASIC_COMPONENT_H_
+#define _BASIC_COMPONENT_H_
+
+#include <sst/core/sst_config.h>
```

Comment:
sst_config.h should not be included in header files (but should be included in implementation files)

Reply:
fixed in 9feecc2